src/main/scala/fs2/interop/cats/cats.scala

```diff
@@ -24,6 +27,15 @@ package object cats extends Instances {
     def runFoldMap[B](f: A => B)(implicit M: Monoid[B]): Free[F, B] =
       self.runFold(M.empty)((b, a) => M.combine(b, f(a)))
 
+    def runGroupByFoldMap[K, B: Monoid](f: A => K)(g: A => B): Free[F, Map[K, B]] =
+      runFoldMap(a => Map(f(a) -> g(a)))
+
+    def runGroupByFoldMonoid[K](f: A => K)(implicit M: Monoid[A]): Free[F, Map[K, A]] =
+      runFoldMap(a => Map(f(a) -> a))
+
+    def runGroupBy[X[_], K](f: A => K)(implicit A: Applicative[X], M: Monoid[X[A]]): Free[F, Map[K, X[A]]] =
```

Comment:
Hm, right - that should be fixed in next cats release. For now, let's implement a local monoid implementation inside the def body of runFoldMap.